wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -382,7 +382,7 @@ public static WUState getStateID(WorkunitInfo wu)
         if (wu != null) {
             if (wu.getStateID() != null) {
                 return getStateID(wu.getStateID());
-            } else if (wu.getState() != null && wu.getState().trim().length() != 0) {
+            } else if (wu.getState() != null) {
```

Comment:
I'm not concerned about a string made up of only whitespace.  It will return UNKNOWN just as if it was empty.  Adding the trim().length() check was my original mistake.